cl/recap/tests/tests.py

```diff
@@ -6155,9 +6149,7 @@ async def test_update_webhook_after_network_error(
 
         with mock.patch(
             "cl.api.webhooks.requests.post",
-            side_effect=lambda *args, **kwargs: exec(
-                "raise ConnectionError('Connection Error')"
-            ),
+            side_effect=requests.ConnectionError("Connection Error"),
```

Comment:
WOW this one took a *long* time to debug.

Ruff rule F401 removed the "unused" import `from requests import ConnectionError`. That didn't cause this exec'd code to fail, though, because Python has a built-in `ConnectionError`: https://docs.python.org/3.13/library/exceptions.html#ConnectionError . So the `exec`'d string returned that instead, leading to a hard-to-debug failure in the test.

The `exec()` was never needed, `side_effect` will raise an exception object that it is given, so that's what this PR does.

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect

😅

Reply:
Oh yeah. That was tricky, thanks for the fix!